.github/workflows/dusk_ci.yml

```diff
@@ -19,73 +18,108 @@ jobs:
         with:
           command: dusk-analyzer
 
-  test_nightly:
-    name: Nightly tests std # This is already testing the `alloc` feature.
+  fmt:
+    name: Check code formatting
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: nightly
+      - run: rustup component add rustfmt
       - uses: actions-rs/cargo@v1
         with:
-          command: test
-          args: --release 
+          command: fmt
+          args: --all -- --check
 
-  test_nightly_canon:
-    name: Nightly tests canon
+  build_docs:
+    name: Build documentation
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: nightly
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: --release --features canon
+      - run: make doc
+      - run: make doc-internal
 
-  build_nightly_nostd:
-    name: Nightly build no_std
+  build_no_std:
+    name: Build no_std
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain:
+          - nightly
+        target: [ thumbv6m-none-eabi ]
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
         with:
-          profile: minimal  
-          toolchain: nightly
-      # ARM contains no std lib. Just core and alloc.
-      - run: rustup target add thumbv6m-none-eabi
-      - uses: actions-rs/cargo@v1
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+
+      - name: Add target
+        run: rustup target add ${{ matrix.target }}
+
+      - name: Build project with alloc
+        uses: actions-rs/cargo@v1
         with:
           command: build
-          args: --release --no-default-features --features alloc --target thumbv6m-none-eabi
-      - uses: actions-rs/cargo@v1
+          args: --release --no-default-features --features alloc --target ${{ matrix.target }}
```

Comment:
If we are just building without any tests, then `release` is just additional overhead that brings no benefit

Reply:
In this case I just kept `release` for the sake of keeping it like it was before. A pro is that it is more consistent with all else that is happening, but I agree right now it adds no tangible benefit.

Reply:
One thing that I would say as well is that the building incurs much less of a cost than running the tests, and since the jobs run in parallel this is not noticeable.